### ISSUE TITLE
File search

### ISSUE THREAD

**Issue by jupe** [silverwind/droppy#125]
It would be very useful if users could search files by filename(/and contents) directly from UI.

**Comment by silverwind** [comment]
I'll just keep a simple directory tree in memory for the first implementation. It can be done pretty efficient as sub-trees don't need to contain the full containing path, and the files can be an array on each node. Objects are pretty well suited, if you think about it.

**Comment by ZombieHippie** [comment]
Here's a script that can take a document of files and convert it to an array of full paths.

It could be used with fuse.js.

http://codepen.io/ZombieHippie/pen/YPNEYe?editors=001
```javascript

function createIndex (files) {
  var result = [];
  function index (prefix, object) {
    var path, filename;
    for (filename in object) {
      path = prefix + "/" + filename
      object[filename] && index(path, object[filename])
      result.push(path)
    }
  }
  index("", files)
  return result
}
```

**Comment by silverwind** [comment]
Exactly what I need (sans semicolons), thanks!

**Comment by ZombieHippie** [comment]
Darn, I should have written it in CoffeeScript!

**Comment by silverwind** [comment]
I heard ECMAScript 8 will finally do away with them, I look forward to that day.

**Comment by silverwind** [comment]
I think I'll provide fuzzy search suggestions through [fuse](https://github.com/krisk/Fuse). Ajax the search string on every keystroke, and return results.

**Comment by ZombieHippie** [comment]
Why wouldn't you use WebSockets?

**Comment by silverwind** [comment]
I literally edited that post a minute after posting :wink:

**Comment by silverwind** [comment]
The search feature is now available since v7.0.0!
